### PR TITLE
WIP Minidlna molecule test

### PR DESCRIPTION
**Pull request by julien-lecomte**
Same as minidlna, but with molecule test

### CODE REVIEW

File: ansible/molecule/bootstrap/molecule.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+#
+# Copyright (C) 2021 Julien Lecomte <julien@lecomte.at>
+# Copyright (C) 2021 DebOps <https://debops.org/>
+# SPDX-License-Identifier: GPL-3.0-only
+#
+
+platforms:
+  - name: debops-apt-stretch
+    image: registry.gitlab.com/dockerio/debian:stretch
```
Comment (julien-lecomte) [inline ansible/molecule/bootstrap/molecule.yml:10]: The docker hub rate limits are easily encountered when all tests "no deps" test are molecule based,

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -33,18 +34,36 @@ variables:
...
+# https://docs.gitlab.com/ee/ci/yaml/README.html#switch-between-branch-pipelines-and-merge-request-pipelines
+workflow:
+  rules:
+    - if: '$CI_COMMIT_BRANCH && $CI_OPEN_MERGE_REQUESTS && $CI_PIPELINE_SOURCE == "push"'
+      when: never
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event"'
+      when: always
+    - if: '$CI_COMMIT_BRANCH && $CI_OPEN_MERGE_REQUESTS'
+      when: never
+    - if: '$CI_COMMIT_BRANCH'
```
Comment (julien-lecomte) [inline .gitlab-ci.yml:57]: Although not necessary, this line was part of the FAQ above, so I'm keeping this.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -33,18 +34,36 @@ variables:
...
+      when: always
+    - if: '$CI_REF_PROTECTED'
```
Comment (julien-lecomte) [inline .gitlab-ci.yml:59]: This is to have a full CI on branches such as master, or stable-x-y. Also, when in GitLab, you can set variable to "1" when running the pipeline manually to have all the jobs.

**Comment by drybjed** [comment]
@julien-lecomte If it's not too much trouble, perhaps you could create two PRs - one for the `minidlna` role, and another for the support for Molecule to run GitLab CI pipeline. That way `minidlna` can be reviewed easier and merged, while the CI pipeline stuff requires more extensive testing. You don't neec to base the CI pipeline explicitly on the `minidlna` role, just start with an existing one, for example `apt`.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -2580,3 +2631,31 @@ stages:
     JANE_INVENTORY_HOSTVARS: 'yadm__dotfiles_enabled=true'
     JANE_DIFF_PATTERN: '.*/roles/yadm/.*'
     JANE_LOG_PATTERN: '\[yadm\]'
+
+
+# === Ansible no dependencies roles === [[[2
+
+'ferm role':
```
Comment (julien-lecomte) [inline .gitlab-ci.yml:2638]: I'm moving the alphabetical list to number of dependencies lists, with inner alphabetical organisation. For some reason I had errors that the `needs` dependencies were not predeclared.

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -2580,3 +2631,31 @@ stages:
...
+  extends: .tpl_molecule_no_deps
+  needs:
+    - 'bootstrap playbook'
+
```
Comment (julien-lecomte) [inline .gitlab-ci.yml:2642]: I can't add rules for now as this would generate errors in the case ferm was not run.

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -2580,3 +2631,31 @@ stages:
...
+'minidlna role':
+  extends: .tpl_molecule_1st_deps
+  needs:
+    - 'ferm role'
+  rules:
+    - if: '$CI_REF_PROTECTED'
+      when: always
+    - if: '$CI_PIPELINE_SOURCE == "merge_request_event" || ($CI_PIPELINE_SOURCE == "push" && $CI_COMMIT_BRANCH)'
+      changes:
+        - ansible/playbooks/service/ferm.yml
```
Comment (julien-lecomte) [inline .gitlab-ci.yml:2654]: The size of this list will be proportional to the number of dependencies.

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -1969,7 +2020,7 @@ stages:
     JANE_LOG_PATTERN: '\[python\]'
 
 'python/raw role':
```
Comment (julien-lecomte) [inline .gitlab-ci.yml:2022]: This job could actually disappear since it's tested via the bootstrap job.

File: .gitlab-ci.yml (hunk 6 above)
Comment (julien-lecomte) [reply .gitlab-ci.yml:2654]: note to self: also dependent on etc_services.

**Pull request closed without merging**